File: tools/graph/fixtures/edge-cases/syntaxError.ts
/**
 * Edge case: TypeScript file with syntax errors.
 * Tests parser error handling and recovery.
 */

export interface ValidInterface {
  name: string;
  age: number;

// SYNTAX ERROR: Missing closing brace above

export type BrokenType = {
  value: string
  // SYNTAX ERROR: Missing semicolon and closing brace

export const malformedObject = {
  key: "value"
  // SYNTAX ERROR: Missing comma
  another: "value"
  // SYNTAX ERROR: Missing closing brace

// SYNTAX ERROR: Unclosed string
export const unclosedString = "This string never ends

// SYNTAX ERROR: Invalid generic syntax
export type BadGeneric<T extends = string;

// SYNTAX ERROR: Mismatched brackets
export function badFunction(): void {
  const arr = [1, 2, 3;
  return;
}}

// SYNTAX ERROR: Invalid type annotation
export const invalidType: : string = "test";

File: tools/graph/fixtures/edge-cases/SyntaxError.cs
namespace Koinon.Domain.Entities;

/// <summary>
/// This file contains intentional syntax errors to test parser error handling.
/// </summary>
public class SyntaxErrorEntity : Entity
{
    // Missing closing brace for property
    public string Name { get; set;

    // Missing semicolon
    public int Age { get; set }

    // Unclosed string literal
    public string Description = "This is unclosed

    // Mismatched braces
    public void DoSomething()
    {
        if (true)
        {
            Console.WriteLine("Test")
        }
    }}

    // Missing type
    public MissingType Value { get; set; }
